app/human-detector-app/src/ble/bleServices.ts

```diff
@@ -0,0 +1,239 @@
+import { BleError, Characteristic, Device, Subscription, BleManager } from 'react-native-ble-plx';
+import NetInfo from '@react-native-community/netinfo';
+import * as EyeSpyUUID from '../../config/BLEConfig';
+import { base64ToJson, isDuplicateDevice, jsonToBase64 } from './helpers';
+
+/**
+ * This file includes all of the camera bluetooth services that the mobile app
+ * will need to connect to.
+ */
+
+export interface CameraSerial {
+  Serial: string;
+  PubKey: string;
+}
+
+export enum WifiSecType {
+  WPA2_802_1X = "KEY_802_1X",
+  WPA2_PSK = "KEY_PSK",
+  OPEN = "OPEN",
+  UNSUPPORTED = "UNSUPPORTED",
+  UNKNOWN = "",
+}
+
+/**
+ * BLE Notification 
+ */
+
+export enum ConnectionStatus {
+  INTERNAL_ERROR = 0,
+  DISCONNECTED = 1,
+  CONNECTING = 2,
+  SUCCESS = 3,
+  FAIL = 4,
+}
+
+// Notification JSON recieved from camera
+export interface ConnectionNotification {
+  State: ConnectionStatus,
+  Reason: number,
+}
+
+export type NotificationCallback = (
+  error: BleError | null,
+  notif: ConnectionNotification | null
+) => Promise<void>;
+
+export class BLEService {
+  private connectedDevice: Device | null = null;
+
+  private devices: Device[] = [];
+
+  private bleManager = new BleManager();
+
+  /**
+   * scanForDevices will start scanning for BLE connections.  It will make it so it only
+   * scans for our specific devices.  It will also add it to a list which can be displayed to the user.
+   */
+  public scanForDevices(callback: (devices: Device[]) => void) {
+    this.bleManager.startDeviceScan([EyeSpyUUID.BLE_SERVICE_UUID], null, (error, device) => {
+      if (error) {
+        console.error(error.errorCode);
+      }
+
+      if (device) {
+        // Add Device
+        if (!isDuplicateDevice(this.devices, device)) {
+          this.devices = [...this.devices, device];
+          callback(this.devices);
+        }
+      }
+    });
+  };
+
+  /**
+   * Returns all scanned devices found
+   * 
+   * @returns Array of scanned devices
+   */
+  public getDevices() {
+    return this.devices;
+  }
+
+  /**
+   * Currently connected device
+   * 
+   * @returns Device
+   */
+  public getCurrentConnectedDevice() {
+    return this.connectedDevice;
+  }
+
+  /**
+   * This method will connect to a specified device and set it
+   * as the currentDevice
+   * @param device
+   */
+  public async connectToDevice(device: Device) {
+    if (await device.isConnected()) return;
+    const deviceConnection = await device.connect();
+    this.connectedDevice = await deviceConnection.discoverAllServicesAndCharacteristics();
+    this.bleManager.stopDeviceScan();
+  };
+
+  /**
+   * Disconnect from current device
+   */
+  public async disconnectFromDevice() {
+    if (this.connectedDevice) {
+      await this.connectedDevice.cancelConnection();
```

Comment:
Why do we need an await here?

Reply:
Didn't want the device to become null until all transactions and stuff were ended, just to be safe.